secure/key-manager/src/lib.rs

```diff
@@ -203,14 +204,34 @@ where
         let storage_key = self.storage.get_public_key(CONSENSUS_KEY)?.public_key;
         COUNTERS.consensus_rotation_tx_resubmissions.inc();
         self.submit_key_rotation_transaction(storage_key)
-            .map(|_| ())
+            .map(|_| ())?;
+        self.submit_reconfiguration_transaction().map(|_| ())
     }
 
     pub fn rotate_consensus_key(&mut self) -> Result<Ed25519PublicKey, Error> {
         let new_key = self.storage.rotate_key(CONSENSUS_KEY)?;
         info!("Successfully rotated the consensus key in secure storage.");
         COUNTERS.completed_consensus_key_rotations.inc();
-        self.submit_key_rotation_transaction(new_key)
+        let result = self.submit_key_rotation_transaction(new_key);
+        self.submit_reconfiguration_transaction()?;
+        result
+    }
+
+    pub fn submit_reconfiguration_transaction(&self) -> Result<(), Error> {
```

Comment:
@davidiw, @valerini how do we feel about adding the "submit_reconfiguration_transaction" logic to the key manager, here? My feeling is that this is a little out of place and breaks the abstractions (given that a key manager only operates on each validator node and won't have access to the association key in production). 

The way that the key manager operates at the moment is that is assumes an external force is responsible for performing reconfigurations. Perhaps this should happen somewhere else? e.g., the management tool?

Reply:
yeah, this isn't good, we'll need to address this, I guess I'll leave it up to you to decide in this PR or another...

Reply:
Indeed, this needs to be moved out! I can put this function in the test.rs. Let me try to do it in the next PR as this PR is becoming too big.